core/client/README.md

```diff
@@ -0,0 +1,364 @@
+
+Alternatively, can also manually edit your `go.mod` and then run `go mod tidy`.
+
+Your `go.mod` should be similar to the following:
+
+```go
+module mykwilapp
+
+go 1.22
+
+require (
+    github.com/kwilteam/kwil-db/core v0.1.2
+)
+```
+
+## Import the `client` package
+
+With the Kwil `core` module added to your `go.mod`, you can use the `client` package in your code by importing it:
+
+```go
+import "github.com/kwilteam/kwil-db/core/client"
+```
+
+## Using the `Client` type
+
+### Basic functionality
+
+The main functionality is provided by the `Client` type. The `NewClient` function constructs a new `Client` instance from the URL of a Kwil RPC provider, and a set of options in the `core/types/client.Options` type.
+
+For example:
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kwilteam/kwil-db/core/client"
+	klog "github.com/kwilteam/kwil-db/core/log"
+	"github.com/kwilteam/kwil-db/core/types"
+	ctypes "github.com/kwilteam/kwil-db/core/types/client"
+)
+
+const (
+	provider = "https://longhorn.kwil.com"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// Create the client and connect to the RPC provider.
+	cl, err := client.NewClient(ctx, provider, ctypes.DefaultOptions())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Report the chain ID and block height of the provider.
+	chainInfo, err := cl.ChainInfo(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("Connected to Kwil chain %q, block height %d\n",
+		chainInfo.ChainID, chainInfo.BlockHeight)
+}
+```
+
+### Wallet Setup
+
+In the above example, we used `ctypes.DefaultOptions()`, which includes no
+logger, signer (wallet), or expected chain ID. To work with a Kwil account, use
+the `crypto` and `crypto/auth` packages to create and load private keys. For
+example, we can create and load a secp256k1 private key plus an Ethereum
+"personal" signer with the following functions:
+
+```go
+import (
+   	"github.com/kwilteam/kwil-db/core/crypto"
+	"github.com/kwilteam/kwil-db/core/crypto/auth"
+)
+
+func genKey() *crypto.Secp256k1PrivateKey {
+	key, _ := crypto.GenerateSecp256k1Key()
+	return key // fmt.Println(key.Hex())
+}
+
+func makeSigner(keyHex string) auth.Signer {
+	key, err := crypto.Secp256k1PrivateKeyFromHex(testKeyHex)
+	if err != nil {
+		panic(fmt.Sprintf("bad private key: %v", err))
+	}
+	return &auth.EthPersonalSigner{Key: *key}
+}
+```
+
+Now we can expand our example application to work with our account and create signed transactions on the specified Kwil network.
+
+```go
+const (
+	chainID  = "longhorn" // expect provider to report this chain ID
+	provider = "https://longhorn.kwil.com"
+	privKey  = "..." // my secp256k1 private key in hexadecimal
+)
+
+func main() {
+	ctx := context.Background()
+	signer := makeSigner(privKey)
+	acctID := signer.Identity()
+
+	opts := &ctypes.Options{
+		Logger:  klog.NewStdOut(klog.InfoLevel),
+		ChainID: chainID, // ensure the provider matches
+		Signer:  signer,  // required for transactions and auth
```

Comment:
Does the signature type need to be specified anywhere? Is this the same process for Ed25519 / or other custom signers?

Reply:
Good question.  The signer itself conveys this.  When `core/transactions.(*Transaction).Sign` call's the signer's `Sign` method, It creates a `Signature` with `Type` set correctly so that it may be verified with the corresponding authenticator.

So when we use `makeSigner`, it creates a `*auth.EthPersonalSigner` that will set it's type as above.  To use ed25519, we'd use this instead to make the `Signer`:

```go
func makeEdSigner(keyHex string) auth.Signer {
	key, err := crypto.Ed25519PrivateKeyFromHex(keyHex)
	if err != nil {
		panic(fmt.Sprintf("bad private key: %v", err))
	}
	return &auth.Ed25519Signer{*key}
}
```

Reply:
Ah perfect - yeah I don't think we need to add that to the docs for now.